classes/class-backup.php

```diff
@@ -917,15 +917,37 @@ public function mysqldump_fallback() {
 	 */
 	public function archive() {
 
-		// Do we have the path to the zip command
-		if ( ( defined( 'HMBKP_FORCE_ZIP_METHOD' ) && ( 'zip' === HMBKP_FORCE_ZIP_METHOD ) ) || $this->get_zip_command_path() ) {
-			$this->zip();
-		}
-		// If not or if the shell zip failed then use ZipArchive
-		elseif ( ( defined( 'HMBKP_FORCE_ZIP_METHOD' ) && ( 'ziparchive' === HMBKP_FORCE_ZIP_METHOD ) ) || ( empty( $this->archive_verified ) && class_exists( 'ZipArchive' ) && empty( $this->skip_zip_archive ) ) ) {
-			$this->zip_archive();
+		if ( defined( 'HMBKP_FORCE_ZIP_METHOD' ) ) {
+			switch ( HMBKP_FORCE_ZIP_METHOD ) {
+				case 'zip':
+					if ( $this->get_zip_command_path() ) {
+						$this->zip();
+					} else {
+						$this->warning( $this->get_archive_method(), __( 'Zip command is not available.', 'backupwordpress' ) );
+					}
+					break;
+				case 'ziparchive':
+					if ( class_exists( 'ZipArchive' ) ) {
+						$this->zip_archive();
+					} else {
+						$this->warning( $this->get_archive_method(), __( 'ZipArchive method is not available.', 'backupwordpress' ) );
+					}
+					break;
+				default:
+					$this->warning( $this->get_archive_method(), __( 'No valid archive method found.', 'backupwordpress' ) );
+					break;
+			}
 		} else {
-			$this->warning( $this->get_archive_method(), __( 'No valid archive method found.', 'backupwordpress' ) );
+			// Is zip available
+			if ( $this->get_zip_command_path() ) {
+				$this->zip();
+			}
+			// If not or if the shell zip failed then use ZipArchive
+			elseif ( empty( $this->archive_verified ) && class_exists( 'ZipArchive' ) && empty( $this->skip_zip_archive ) ) {
```

Comment:
Now that we don't support PclZip I think we can drop `skip_zip_archive` completely?

Also I think there's an issue here with using `elseif`, we want to fallback to `ZipArchive` if `zip` doesn't work for whatever reason, even if `zip_command_path` is valid. Currently we'll never fallback to `ZipArchive` because of the `elseif`.